### PR TITLE
Dev/kapish patel/creating schemas

### PR DESCRIPTION
**Pull request by kapish-patel**
I have created some function in models and made some new routes check and let me know change if any.

### CODE REVIEW

**Review by vidhireems — APPROVED** [review]

File: interfaces/IPremCustomerModel.ts [hunk 1]
```diff
@@ -4,7 +4,6 @@ import Mongoose from 'mongoose';
 //Interface for restaurant model
 interface IPremCustomerModel extends Mongoose.Document {
     id: String;
-    isValeParking: Boolean;
```
Comment (Harshithamlinganna) [inline interfaces/IPremCustomerModel.ts:7]: can you rename it isValetParking ?

File: interfaces/IUserModel.ts [hunk 2]
```diff
@@ -3,11 +3,11 @@ import Mongoose from 'mongoose';
 
 //Interface for restaurant model
 interface IUserModel extends Mongoose.Document {
-    id: String;
+    userId: String;
     name: String;
-    Email: String;
```
Comment (Harshithamlinganna) [inline interfaces/IUserModel.ts:8]: Why do we have two ids here?
Can you appropriately name them

File: models/CustomerUserModel.ts [hunk 3]
```diff
@@ -69,34 +70,56 @@ class CustomerUserModel {
...
-            const { address, contactNumber } = request.body;
-            if (!address || !contactNumber) {
+            const { address, contactNumber, name, email, password } = request.body;
+            if ( !address || !contactNumber || !name || !email || !password || !customerId ) {
                 return response.status(400).json({ message: "Please fill all fields" });
             }
-            const customer = new this.model({
-                customerId,
-                address,
-                contactNumber,
```
Comment (Harshithamlinganna) [inline models/CustomerUserModel.ts]: We can add a required constraint to the attributes in a scheme.

File: models/PremCustomerModel.ts [hunk 4]
```diff
@@ -23,8 +23,7 @@ class PremCustomerModel {
         this.schema = new Mongoose.Schema(
             {
                 id: String,
-                isValeParking: Boolean,
-                tableNumber: Number,
+                tableNumber: Number,  
                 numberOfPeople: Number,
             }, {collection: 'PremCustomer'}
         );
```
Comment (Harshithamlinganna) [inline models/PremCustomerModel.ts:29]: To make the code more readable, PremCustomer can be named to PremiumCustomer

File: models/UserModel.ts [hunk 5]
```diff
@@ -24,12 +24,12 @@ class UserModel {
...
                 email: String,
-                Password: String,
-                User_Type: String,
-                Refrence_User_Type_Id: String,
+                password: String,
+                userType: String,
+                refrenceUserTypeId: String,
             }, {collection: 'Users'}
         );
     }
```
Comment (Harshithamlinganna) [inline models/UserModel.ts:35]: the user model here is the restaurant user?,, if it is just the parent model for the customer model we don't need this   .. if it is  for later usecase, I would suggest keeping the name appropriate for readability.

**Review by Harshithamlinganna — CHANGES_REQUESTED** [review]

File: interfaces/IPremCustomerModel.ts (hunk 1 above)
Comment (kapish-patel) [reply interfaces/IPremCustomerModel.ts:7]: as we discussed we are not currently performing valet parking feature so i have removed "ValeParking " on line 7 from the code not sure where is came from it not there in my branch

File: interfaces/IUserModel.ts (hunk 2 above)
Comment (kapish-patel) [reply interfaces/IUserModel.ts:8]: as above this is also resolved but nit sure how come there are some duplicates currently this mode look like this
image

File: models/PremCustomerModel.ts (hunk 4 above)
Comment (kapish-patel) [reply models/PremCustomerModel.ts:29]: copy that

File: models/UserModel.ts (hunk 5 above)
Comment (kapish-patel) [reply models/UserModel.ts:35]: yes this is user model which will server 2 other model restaurant employee and customer if you are still confused you can look up into the class diagram.

File: interfaces/IPremCustomerModel.ts (hunk 1 above)
Comment (kapish-patel) [reply interfaces/IPremCustomerModel.ts:7]: image

**Pull request merged**